src/Http/Client/Response.php

```diff
@@ -401,12 +403,14 @@ protected function _getCookies(): array
     {
         $this->buildCookieCollection();
 
-        $cookies = [];
-        foreach ($this->cookies as $cookie) {
-            $cookies[$cookie->getName()] = $this->convertCookieToArray($cookie);
+        $out = [];
+        /** @var \Cake\Http\Cookie\Cookie[] $cookies */
```

Comment:
Isn't $this->cookies a CookieCollection?

Reply:
Exactly, the iterating over it does provide the Cookie objects.
A bit too much hidden magic for my taste, but thats how this works here and also in other collections in Cake.

Reply:
The correct approach would be this: https://github.com/cakephp/cakephp/pull/13223/commits/4bd026a0f716f1767409c213f1761bf864b88b6c#diff-6d806173a45816b0375aaaece3a987a7R98
But apparently, some static analyzers are not happy with this.